src/System.Collections/tests/BitArray/BitArray_GetSetTests.cs

```diff
@@ -307,5 +307,77 @@ public static void SyncRoot()
             Assert.Same(bitArray.SyncRoot, bitArray.SyncRoot);
             Assert.NotSame(bitArray.SyncRoot, ((ICollection)new BitArray(10)).SyncRoot);
         }
+
+        public static IEnumerable<object> CopyTo_Hidden_Data()
+        {
+            yield return new object[] { "ZeroLength", new BitArray(0) };
+            yield return new object[] { "Constructor", new BitArray(BitsPerInt32 / 2 - 3, true) };
+            yield return new object[] { "Not", new BitArray(BitsPerInt32 / 2 - 3, false).Not() };
+            BitArray setAll = new BitArray(BitsPerInt32 / 2 - 3, false);
+            setAll.SetAll(true);
+            yield return new object[] { "SetAll", setAll };
+            BitArray lengthShort = new BitArray(BitsPerInt32, true);
+            lengthShort.Length = BitsPerInt32 / 2 - 3;
+            yield return new object[] { "Length-Short", lengthShort };
+            BitArray lengthLong = new BitArray(2 * BitsPerInt32, true);
+            lengthLong.Length = BitsPerInt32 - 3;
+            yield return new object[] { "Length-Long < 32", lengthLong };
+            BitArray lengthLong2 = new BitArray(2 * BitsPerInt32, true);
+            lengthLong2.Length = BitsPerInt32 + 3;
+            yield return new object[] { "Length-Long > 32", lengthLong2 };
+            // alligned test cases
+            yield return new object[] { "Aligned-Constructor", new BitArray(BitsPerInt32, true) };
+            yield return new object[] { "Aligned-Not", new BitArray(BitsPerInt32, false).Not() };
+            BitArray alignedSetAll = new BitArray(BitsPerInt32, false);
+            alignedSetAll.SetAll(true);
+            yield return new object[] { "Aligned-SetAll", alignedSetAll };
+            BitArray alignedLengthLong = new BitArray(2 * BitsPerInt32, true);
+            yield return new object[] { "Aligned-Length-Long", alignedLengthLong };
+        }
+
+        [Theory]
+        [MemberData(nameof(CopyTo_Hidden_Data))]
+        public static void CopyTo_Int_Hidden(string label, BitArray bits)
+        {
+            int[] data = new int[2];
+            ((ICollection)bits).CopyTo(data, 0);
+
+            int maxValue = unchecked(((int)0xffffffff));
+            int fullInts = bits.Length / BitsPerInt32;
+            int remainder = bits.Length % BitsPerInt32;
+
+            for (int i = 0; i < fullInts; ++i)
```

Comment:
It can be _extremely_ difficult to tell which entry fails when the iteration is external to the assert.  You should be using `Assert.All` here:
```
Assert.All(data.Take(fullInts), d => Assert.Equal(maxValue, d));
```